From da8b3138d3baf4ffcf1ed807b63bfba4e80651b8 Mon Sep 17 00:00:00 2001
From: Shinobu Uehara <shinobu.uehara.xc@renesas.com>
Date: Fri, 14 Jun 2013 15:30:04 +0900
Subject: [PATCH 517/715] mmc: core: Fix block size after password protection

The block size is returned to 512 byte.

Signed-off-by: Shinobu Uehara <shinobu.uehara.xc@renesas.com>
(cherry picked from commit 93bc8e894a428445206115457bfd02145e0a8b0f)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 drivers/mmc/core/mmc_ops.c |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/mmc/core/mmc_ops.c b/drivers/mmc/core/mmc_ops.c
index 8be3fcd..b91ea23 100644
--- a/drivers/mmc/core/mmc_ops.c
+++ b/drivers/mmc/core/mmc_ops.c
@@ -1,6 +1,7 @@
 /*
  *  linux/drivers/mmc/core/mmc_ops.h
  *
+ * Copyright (C) 2013 Renesas Electronics Corporation
  *  Copyright 2006-2007 Pierre Ossman
  * MMC password protection (C) 2006 Instituto Nokia de Tecnologia (INdT),
  * All Rights Reserved.
@@ -155,6 +156,15 @@ int mmc_lock_unlock(struct mmc_card *card, struct key *key, int mode)
 	else
 		card->state &= ~MMC_STATE_LOCKED;
 
+	memset(&cmd, 0, sizeof(struct mmc_command));
+
+	cmd.opcode = MMC_SET_BLOCKLEN;
+	cmd.arg = 512;
+	cmd.flags = MMC_RSP_R1 | MMC_CMD_AC;
+	err = mmc_wait_for_cmd(card->host, &cmd, MMC_CMD_RETRIES);
+	if (err)
+		goto out;
+
 out:
 	kfree(data_buf);
 
-- 
1.7.10.4

